src/runtime/CMakeLists.txt

```diff
@@ -34,8 +34,29 @@ target_include_directories(hipSYCL-rt
     ${PROJECT_BINARY_DIR}/include
 )
 
+# to get the search path next to the rt lib, we need to know the output name, so set it explicitly.
+set(HIPSYCL_RT_LIBRARY_OUTPUT_NAME "hipSYCL-rt")
+if(WIN32)
+  set_target_properties(hipSYCL-rt PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS On)
+
+  # ensure actual output and expected are equivalent
+  set_target_properties(hipSYCL-rt PROPERTIES RUNTIME_OUTPUT_NAME ${HIPSYCL_RT_LIBRARY_OUTPUT_NAME})
+  set(HIPSYCL_RT_LIBRARY_OUTPUT_NAME "${HIPSYCL_RT_LIBRARY_OUTPUT_NAME}.dll")
+else()
+  target_link_libraries(hipSYCL-rt PRIVATE dl)
+
+  # ensure actual output and expected are equivalent
+  set_target_properties(hipSYCL-rt PROPERTIES LIBRARY_OUTPUT_NAME ${HIPSYCL_RT_LIBRARY_OUTPUT_NAME})
+  set(HIPSYCL_RT_LIBRARY_OUTPUT_NAME "lib${HIPSYCL_RT_LIBRARY_OUTPUT_NAME}.so")
+endif()
+
+configure_file(
+  ${PROJECT_SOURCE_DIR}/include/hipSYCL/common/config.hpp.in
+  ${PROJECT_BINARY_DIR}/include/hipSYCL/common/config.hpp)
```

Comment:
This file is not only required by the runtime. Why does it need to be configured by the runtime CMakeLists.txt? Can we not also configure in the main CMakeLists?

Reply:
We need the `HIPSYCL_RT_LIBRARY_OUTPUT_NAME` for the config as well, which was not available in the root.
The variable is currently required on Windows for the `GetModuleHandle`, as I did not want to hardcode it in there and to make sure it actually fits the output name, I added this mechanism here.

Reply:
Tried mitigating this in 257f63c